09_employment/employment_metrics_county_all.qmd

```diff
@@ -0,0 +1,856 @@
+
+```{r}
+count(acs_crosswalked, afact == 0)
+
+acs_crosswalked <- acs_crosswalked %>% 
+  filter(afact > 0)
+
+```
+
+Adjust the person weight to account for PUMA-to-county mapping (those where the PUMA is not entirely inside the county).
+
+```{r}
+select(acs_crosswalked, perwt, repwtp1, repwtp80, afact)
+
+acs_crosswalked <- acs_crosswalked %>%
+  mutate(perwt = perwt * afact) %>% 
+  mutate(across(matches("repwtp[0-9]+"), ~.x * afact))
+
+select(acs_crosswalked, perwt, repwtp1, repwtp80, afact) 
+
+```
+
+Create a binary variable to distinguish between 1 and 5 year ACS surveys. 5-year samples include the term "5-year".
+
+```{r}
+acs_crosswalked <- acs_crosswalked %>%
+  mutate(sample_structure = ifelse(str_detect(sample, "5-year"), "5year", "1year")) 
+
+```
+
+Create a binary employed variable for calculating the share employed. IPUMS data codes "Employed" as equal to 1.
+
+```{r}
+acs_crosswalked <- acs_crosswalked %>% 
+  mutate(employed = employment_status == 1) 
+
+```
+
+Check how many records, if any, have NA reported for `employment_status` after filtering (this should be zero).
+
+```{r}
+count(acs_crosswalked, is.na(employment_status))
+
+```
+
+Confirm there are no vacant properties included in the data.
+
+```{r}
+acs_crosswalked %>%
+  filter(is.na(vacancy)) %>% 
+  count(pernum)
+
+```
+
+## Create employment metric: All
+
+### Pre-drop suppressed cells
+
+First, to save time when processing the metric, calculate the effective sample count for each county, defined as the sum of the `afact` variable. This is used to create a cutoff for data quality based on the actual number of survey records being used in the calculation. We will not include counties with less than 30 effective samples.
+
+```{r}
+#| label: all-effective-samples
+
+acs_all <- acs_crosswalked %>%
+  filter(sample_structure == "1year") %>% 
+  group_by(year, sample, crosswalk_period, statefip, county) %>%
+  mutate(effective_sample = sum(afact)) %>% 
+  ungroup()
+
+# calculate the number of statistics after suppressing values
+acs_all %>%
+  group_by(year, sample, crosswalk_period, statefip, county) %>%
+  summarize(
+    effective_sample = max(effective_sample)
+  ) %>%
+  ungroup() %>%
+  summarize(
+    original_stats = n(),
+    unsuppressed_stats = sum(effective_sample >= 30)
+  )
+
+```
+
+Create the employment metric. 
+
+Objective: get the percent of individuals between the ages 25 and 54 that are employed from the EMPSTAT variable.
+
+Aggregation should be weighted by PERWT (this is a person level statistic).
+
+Check national employment ratio. Compute the ratio (share employed) for 2018, 2021 and 2022 nationally and show results.
+
+Remove records from counties below the size quality cutoff. 
+
+```{r}
+#| label: subset-all
+
+acs_all <- acs_all %>% 
+  filter(effective_sample >= 30)
+
+```
+
+### Calculate metric
+
+Calculate the county-level metrics. 
+
+```{r}
+#| label: calc-all
+
+results_all <- acs_all %>% 
+  group_split(statefip) %>% 
```

Comment:
yes it helps avoid an overload to the system as well because srvyr datasets are very large. This makes is so that only one subsetted srvyr dataset exists at a time.